### DISCUSSION TITLE
Should cask that auto_updates be stored in a directory with their version number?

### DISCUSSION THREAD

**Discussion by bramp** [Homebrew/discussions#4742] [Casks]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

**Question:** I have installed the [flutter cask](https://formulae.brew.sh/cask/flutter). Home brew stored flutter in a directory, `/opt/homebrew/Caskroom/flutter/3.10/flutter/bin/flutter`, but this cask has the `auto_updates true`, and will auto-update itself (well techininically update when I run `flutter upgrade`). When it does auto-update, the updates get stored in the same directory (e.g `/opt/homebrew/Caskroom/flutter/3.10/flutter`).

Now it confused me, because I'm now running Flutter `3.13.1`, however, it's still in the `/opt/homebrew/Caskroom/flutter/3.10/` directory (note the `3.10`).

Should Casks that auto-update store themselves in a versioned directory? or something like `/latest/`?

### Relevant casks

flutter

**Comment by SMillerDev** [comment]
You can still force an update so yes, it should keep track of what was installed.

**Reply by bramp** [reply]
Ok thanks. Then how do we avoid the confusion of `flutter upgrade` writing a new version into a directory with the old version number in the path?

Btw I noticed this because my IDE explicitly shows the full path to the libraries, and flutter looked wrong.

**Reply by SMillerDev** [reply]
You don't, the only way would be to avoid `flutter upgrade` and use `brew upgrade --greedy --cask flutter` instead.

**Reply by bramp** [reply]
That's fair. 

I wonder if there is a way to set `auto_updates false` atleast locally, or ensure this package gets updated with a normal `brew upgrade` without having to specify `--greedy`?
